src/color_laws.jl

```diff
@@ -228,3 +230,141 @@ function gcc09_invum(x::Real, Rv::Real)
 
     return a + b / Rv
 end
+
+# Shape models used by F99 and F04
+function _curve_F99_method(
+    x::Real,
+    Rv::Real,
+    c1::Real,
+    c2::Real,
+    c3::Real,
+    c4::Real,
+    x0::Real,
+    gamma::Real,
+    optnir_axav_x::Vector{<:Real},
+    optnir_axav_y::Vector{<:Real},
+    )
+
+    # x value above which FM90 parametrization used
+    x_cutval_uv = 10000.0 / 2700.0
+
+    # required UV points for spline interpolation
+    x_splineval_uv = 10000.0 ./ [2700.0, 2600.0]
+
+    # add in required spline points, otherwise just spline points
+    if x >= x_cutval_uv
+        xuv = vcat(x_splineval_uv, x)
+    else
+        xuv = x_splineval_uv
+    end
+
+    # FM90 model and values
+    fm90_model = FM90(c1=c1, c2=c2, c3=c3, c4=c4, x0=x0, gamma=gamma)
+    # evaluate model and get results in A(x)/A(V)
+    axav_fm90 = fm90_model.(10000.0 ./ xuv) / Rv .+ 1.0 # Expects xuv in Å
+
+    # ignore the spline points
+    if x >= x_cutval_uv
+        axav = last(axav_fm90)
+    else
+        # **Optical Portion**
+        # using cubic spline anchored in UV, optical, and IR
+
+        # optical/NIR points in input x
+
+        # save spline points
+        y_splineval_uv = axav_fm90
+
+        # spline points
+        x_splineval_optir = optnir_axav_x
+
+        # determine optical/IR values at spline points
+        y_splineval_optir = optnir_axav_y
+        spline_x = vcat(x_splineval_optir, x_splineval_uv)
+        spline_y = vcat(y_splineval_optir, y_splineval_uv)
+        spl = Spline1D(spline_x, spline_y, k=3)
+        axav = spl(x)
+    end
+
+    # return A(x)/A(V)
+    return axav
+end
+
+"""
+    F99(;Rv=3.1)
+
+Fitzpatrick (1999) dust law.
+
+# References
+[Fitzpatrick (1999)](https://ui.adsabs.harvard.edu/abs/1999PASP..111...63F/)
+"""
```

Comment:
How about a line describing more about the model (e.g. where is the model applicable)? Check the documentation of other models for reference.

Reply:
Ok, should be up. f458a34791e3c16be89d7c6146c5259988fe80da